pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -615,23 +615,21 @@ func parseDeviceAddress(addrString string) []string {
 	}
 	return addrs
 }
-func (l *LibvirtDomainManager) SyncVMI(vmi *v1.VirtualMachineInstance, useEmulation bool, options *cmdv1.VirtualMachineOptions) (*api.DomainSpec, error) {
-	l.domainModifyLock.Lock()
-	defer l.domainModifyLock.Unlock()
+
+func (l *LibvirtDomainManager) generateConverterContext(vmi *v1.VirtualMachineInstance, useEmulation bool, options *cmdv1.VirtualMachineOptions, isMigrationTarget bool) (*converter.ConverterContext, error) {
```

Comment:
This function has been extracted and shared between two callers, however, each of the callers have subtle differences.
The problem and risk I see here is that the changes include both "moving" the code and changing its content at the same time, allowing little bugs to get in.

For example, `isBlockPVC` is used differently in the two callers, mainly because one did not treated the `hotplugVolumes`. The condition there seems complex and I am not sure if it makes sense.

I would suggest to make the two code blocks (almost) identical before the extraction, reducing the risk of missing out such things (and explaining why they are ok if changed).

Reply:
> I would suggest to make the two code blocks (almost) identical before the extraction, reducing the risk of missing out such things (and explaining why they are ok if changed).

i agree this can improve. i think this is something someone can follow up on later though. this PR is already huge.